source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -276,7 +278,8 @@ void RedisCluster::RedisDiscoverySession::startResolveRedis() {
     const int rand_idx = parent_.random_.random() % discovery_address_list_.size();
     auto it = discovery_address_list_.begin();
     std::next(it, rand_idx);
-    host = Upstream::HostSharedPtr{new RedisHost(parent_.info(), "", *it, parent_, true)};
+    host = Upstream::HostSharedPtr{
```

Comment:
nit: while you are in here make_shared

Reply:
unfortunately cannot easily switch to make_shared here, as one of RedisHost base classes has std::atomic field, thus not copyable in make_shared unless custom copy constructor is provided.